app/js/trouble.js

```diff
@@ -53,15 +47,15 @@
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
+				for (var i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this does remove "making" a function in a loop, invoking the function here will run it immediately when the code is executed rather than when the timeout event fires later. Two approaches can work here: 1. move the function declaration/anonymous function variable assignment outside the loop and pass in a reference to the function as the first argument to setTimeout  or 2. remove the infinite loop and just call setTimeout once.